app/src/main/java/com/example/bookwormadventuresdeluxe2/CreateAccountActivity.java

```diff
@@ -95,54 +100,58 @@ public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth)
-                editTextEmail.setError(null);
-                editTextPassword.setError(null);
-                confirmPassword.setError(null);
-
-                /* Check for Empty EditTexts */
-                if (!TextUtils.isEmpty(editTextEmail.getText().toString())
-                        && !TextUtils.isEmpty(editTextUsername.getText().toString())
-                        && !TextUtils.isEmpty(editTextPhoneNumber.getText().toString())
-                        && !TextUtils.isEmpty(editTextPassword.getText().toString())
-                        && !TextUtils.isEmpty(confirmPassword.getText().toString()))
-                {
-                    /* Check if passwords match */
-                    if (EditTextValidator.passwordsMatch(editTextPassword, confirmPassword))
-                    {
-                        String email = editTextEmail.getText().toString().trim();
-                        String password = editTextPassword.getText().toString().trim();
-                        String username = editTextUsername.getText().toString().trim();
-                        String phoneNumber = editTextPhoneNumber.getText().toString().trim();
+                Boolean noEmpties = true;
 
-                        /* Create user if username is not already taken*/
-                        checkUsernameAvailability(email, password, username, phoneNumber);
-                    }
+                /* Set Empty EditText Error codes */
+                if (TextUtils.isEmpty(confirmPassword.getText().toString().trim()))
+                {
+                    EditTextValidator.isEmpty(confirmPassword);
+                    noEmpties = false;
                 }
-                else
+                if (TextUtils.isEmpty(editTextPassword.getText().toString().trim()))
                 {
-                    /* Set Empty EditText Error code */
-                    if (TextUtils.isEmpty(confirmPassword.getText().toString()))
-                    {
-                        EditTextValidator.isEmpty(confirmPassword);
-                    }
-                    if (TextUtils.isEmpty(editTextPassword.getText().toString()))
-                    {
-                        EditTextValidator.isEmpty(editTextPassword);
-                    }
-                    if (TextUtils.isEmpty(editTextPhoneNumber.getText().toString()))
-                    {
-                        EditTextValidator.isEmpty(editTextPhoneNumber);
-                    }
-                    if (TextUtils.isEmpty(editTextEmail.getText().toString()))
-                    {
-                        EditTextValidator.isEmpty(editTextEmail);
-                    }
-                    if (TextUtils.isEmpty(editTextUsername.getText().toString()))
+                    EditTextValidator.isEmpty(editTextPassword);
+                    noEmpties = false;
+                }
+                if (TextUtils.isEmpty(editTextPhoneNumber.getText().toString().trim()))
+                {
```

Comment:
we can move this to another method, to make onClick easier to follow and make that function return a boolean which would replace "noEmpties"

Reply:
moved to EditText validator